### PR TITLE
Pathd pcep pceplib timers sockets

### PR DESCRIPTION
**Pull request by bradyallenjohnson**

### CODE REVIEW

File: pathd/path_pcep_controller.c [hunk 1]
```diff
@@ -397,15 +442,15 @@ int pcep_thread_send_report_callback(struct thread *t)
...
+int schedule_thread_timer(struct ctrl_state *ctrl_state, int pcc_id,
+			  enum pcep_ctrl_timer_type type, uint32_t delay,
+			  void *payload, struct thread **thread)
 {
 	assert(thread != NULL);
 
 	struct pcep_ctrl_timer_data *data;
 
-	data = XCALLOC(MTYPE_PCEP, sizeof(*data));
+	data = XCALLOC(MTYPE_PCEP, sizeof(struct pcep_ctrl_timer_data));
```
Comment (sylane) [inline pathd/path_pcep_controller.c:453]: Why this change ? I feel that duplicating the type is not an advantage here.

File: pathd/path_pcep_controller.c [hunk 2]
```diff
@@ -473,6 +521,78 @@ int pcep_thread_timer_poll(struct ctrl_state *ctrl_state)
...
+
+int schedule_thread_socket(struct ctrl_state *ctrl_state, int pcc_id,
+			   enum pcep_ctrl_socket_type type, bool is_read,
+			   void *payload, int fd, struct thread **thread)
+{
+	assert(thread != NULL);
+
+	struct pcep_ctrl_socket_data *data;
+
+	data = XCALLOC(MTYPE_PCEP, sizeof(struct pcep_ctrl_socket_data));
```
Comment (sylane) [inline pathd/path_pcep_controller.c:534]: Same here, what is the logic behind repeating the type ?

File: pathd/path_pcep_lib.c [hunk 3]
```diff
@@ -54,28 +66,41 @@ static void free_counter(struct counter *counter);
...
+		.malloc_func = (pceplib_malloc_func)qmalloc,
+		.calloc_func = (pceplib_calloc_func)qcalloc,
+		.realloc_func = (pceplib_realloc_func)qrealloc,
+		.strdup_func = (pceplib_strdup_func)qstrdup,
+		.free_func = (pceplib_free_func)qfree,
+		/* Timers infrastructure */
+		.external_infra_data = fpt,
+		.timer_create_func = pcep_lib_pceplib_timer_create_cb,
+		.timer_cancel_func = pcep_lib_pceplib_timer_cancel_cb,
+		/*.timer_create_func   = NULL,
```
Comment (sylane) [inline pathd/path_pcep_lib.c:91]: What are these comments for ?

Comment (bradyallenjohnson) [reply pathd/path_pcep_lib.c:91]: I will remove these comments, I had them in for testing purposes.

File: pathd/path_pcep_controller.c (hunk 1 above)
Comment (bradyallenjohnson) [reply pathd/path_pcep_controller.c:453]: I had some memory problems, and was trying different things. I'll put it back.

File: pathd/path_pcep_controller.c [hunk 4]
```diff
@@ -316,10 +313,39 @@ void pcep_thread_schedule_reconnect(struct ctrl_state *ctrl_state, int pcc_id,
 	uint32_t delay = backoff_delay(MAX_RECONNECT_DELAY, 1, retry_count);
 	PCEP_DEBUG("Schedule reconnection in %us (retry %u)", delay,
 		   retry_count);
-	schedule_thread(ctrl_state, pcc_id, TM_RECONNECT_PCC, delay, NULL,
-			thread);
+	schedule_thread_timer(ctrl_state, pcc_id, TM_RECONNECT_PCC, delay, NULL,
+			      thread);
+}
+
+void pcep_thread_schedule_pceplib_timer(struct ctrl_state *ctrl_state,
```
[the remark this answers is not in the document]
Comment (bradyallenjohnson) [reply pathd/path_pcep_controller.c:320]: We could, but then we would need to make schedule_thread_timer_with_cb() and schedule_thread_timer() public and accessible from outside path_pcep_controller, and it seems like you wanted that encapsulated.

[the remark this answers is not in the document]
Comment (sylane) [reply pathd/path_pcep_controller.c:320]: ok, it is fine
